tests/distributions/test_multivariate.py

```diff
@@ -340,3 +340,273 @@ def test_check_numerics(self):
+                                         "log\(temperature\).*Tensor had Inf"):
+                log_p.eval(feed_dict={tau: 0., logits: np.ones([2]),
+                                      given: [1., 1.]})
+            with self.assertRaisesRegexp(tf.errors.InvalidArgumentError,
+                                         "log\(temperature\).*Tensor had NaN"):
+                log_p.eval(feed_dict={tau: -1., logits: np.ones([2]),
+                                      given: [1., 1.]})
+
+
+class TestConcrete(tf.test.TestCase):
+    def test_init_check_shape(self):
+        with self.test_session(use_gpu=True):
+            with self.assertRaisesRegexp(ValueError, "should have rank"):
+                Concrete(1., logits=tf.zeros([]))
+
+    def test_init_n_categories(self):
+        con = Concrete(1., tf.ones([10]))
+        self.assertTrue(isinstance(con.n_categories, int))
+        self.assertEqual(con.n_categories, 10)
+
+        with self.test_session(use_gpu=True):
+            logits = tf.placeholder(tf.float32, None)
+            con2 = Concrete(1., logits)
+            self.assertEqual(
+                con2.n_categories.eval(feed_dict={logits: np.ones([10])}), 10)
+            with self.assertRaisesRegexp(tf.errors.InvalidArgumentError,
+                                         "should have rank"):
+                con2.n_categories.eval(feed_dict={logits: 1.})
+
+    def test_init_temperature(self):
+        with self.assertRaisesRegexp(ValueError,
+                                     "should be a scalar"):
+            Concrete([1.], [1., 2.])
+
+        with self.test_session(use_gpu=True):
+            temperature = tf.placeholder(tf.float32, None)
+            con = Concrete(temperature, [1., 2.])
+            with self.assertRaisesRegexp(tf.errors.InvalidArgumentError,
+                                         "should be a scalar"):
+                con.temperature.eval(feed_dict={temperature: [1.]})
+
+    def test_value_shape(self):
+        # static
+        con = Concrete(1., tf.placeholder(tf.float32, [None, 10]))
+        self.assertEqual(con.get_value_shape().as_list(), [10])
+
+        # dynamic
+        logits = tf.placeholder(tf.float32, None)
+        con2 = Concrete(1., logits)
+        self.assertTrue(con2._value_shape().dtype is tf.int32)
+        with self.test_session(use_gpu=True):
+            self.assertEqual(con2._value_shape().eval(
+                feed_dict={logits: np.ones([2, 1, 3])}).tolist(), [3])
+
+        self.assertEqual(con._value_shape().dtype, tf.int32)
+
+    def test_batch_shape(self):
+        def _proxy_distribution(logits):
+            return Concrete(1., logits)
+        utils.test_batch_shape_1parameter(
+            self, _proxy_distribution, np.zeros, is_univariate=False)
+
+    def test_sample_shape(self):
+        def _proxy_distribution(logits):
+            return Concrete(1., logits)
+        utils.test_sample_shape_1parameter_multivariate(
```

Comment:
no I didn't mean that way. I mean from the name of test_sample_1parameter_univariate should be applicable to Categorical but in fact it cannot. So I think the name here is a little confusing..